enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,494 @@
+module](https://github.com/coreos/fedora-coreos-config/pull/346) which will
+apply these settings in the initramfs.
+
+These enhancements provide some excellent improvements to early network
+configuration capabilities.  However, it is currently most beneficial to the
+UPI bare metal flow.  The IPI bare metal flow will need to do more work to
+figure out how to make use of these tools.  Bare Metal IPI makes use of a
+different deployment ramdisk that is part of the Ironic project.  While
+conceptually similar, it has some significant differences in implementation.
+Future integration work may provide better alignment between the Bare Metal IPI
+tooling and the RHCOS installer.
+
+### SR-IOV Operator
+
+Some clusters may want to allocate SR-IOV devices to workloads for a secondary
+network interface with high performance.  These devices are managed by the
+[SR-IOV Operator](https://github.com/openshift/sriov-network-operator).
+
+### Privileged Pods
+
+A use case came up where someone wanted to run a custom BGP routing daemon on
+each host.  Previously they would have managed this directly via Ansible and
+wondered how such a configuration could be done with OCP 4.  This sort of
+custom networking setup is still possible if run in containers.  For example,
+if containers are built including the required daemon(s), then a `DaemonSet`
+could be applied to the cluster to run a privileged pod on each Node which
+performs the desired host network management.
+
+Care must be taken when taking an approach like this.  It’s critical to
+understand how such an approach will interact with the network provider in use
+to ensure unintended side effects do not occur.  While this capability is
+perfectly valid to use, it should not be encouraged.  Instead, it is a
+capability we keep “in our back pocket” in case it’s needed to help a given
+user solve their unique requirements.
+
+## Future Improvements
+
+### Disabling initramfs Networking
+
+Configuration of initramfs networking is lacking in some desired flexibility,
+as shown by the discussion in [this ignition
+issue](https://github.com/coreos/ignition/issues/979).  The default behavior in
+RHCOS is to attempt to get a DHCPv4 or DHCPv6 lease on one of the host’s
+interfaces.  This can cause a lot of undesirable delay if there are a lot of
+network interfaces present and DHCP is not available.  It will also cause
+booting to fail if DHCP never succeeds.
+
+One improvement is the ability to disable initramfs networking completely,
+which would be useful in combined with providing the host a full ignition
+configuration without needing to retrieve the bulk of it from a network
+location.  See “Flattened Ignition Configuration” for some more info on that
+piece.
+
+### NetworkManager in initramfs
+
+Fedora CoreOS has already moved to running NetworkManager in the initramfs.
+More info can be found in
+<https://github.com/coreos/fedora-coreos-tracker/issues/394>.
```

Comment:
This should go away now since you already talk about NM in the initramfs in 4.6 above?

Reply:
good catch, thanks